test/library/middleware-test.ts

```diff
@@ -253,6 +253,23 @@ describe('middleware test', () => {
 		await request(app).get('/').expect(429, message)
 	})
 
+	it('should allow responding with a return value via the function message', async () => {
+		const _message = 'too many requests'
+		const message = async (_request: Request, _response: Response) => {
+			return 'too many requests'
+		}
+
+		const app = createServer(
+			rateLimit({
+				message,
+				max: 1,
+			}),
+		)
+
+		await request(app).get('/').expect(200, 'Hi there!')
+		await request(app).get('/').expect(429, _message)
```

Comment:
This is just a nitpick, but I would probably make this a hard-coded string and get rid of the `_message` variable entirely.